src/Microsoft.AspNetCore.OData/Routing/ODataActionSelector.cs

```diff
@@ -80,10 +81,23 @@ public ActionDescriptor SelectBestCandidate(RouteContext context, IReadOnlyList<
         {
             RouteData routeData = context.RouteData;
             ODataPath odataPath = context.HttpContext.ODataFeature().Path;
+
+            var routers = routeData.Routers;
+            var odataRoute = routeData.Routers.FirstOrDefault((router) =>
+            {
+                var _odataRoute = router as ODataRoute;
+                return _odataRoute == null ?
+                    false : true;
+            }) as ODataRoute;
```

Comment:
You can accomplish it:
1) routeData.Routers.OfType<ODataRoute>().FirstOrDefault()
2) routeData.Routers.FirstOrDefault(r => r is ODataRoute) as ODataRoute;